### PR TITLE
repo2docker setup

### PR DESCRIPTION
**Pull request by rkevin-arch**
Full repo2docker setup, where all the packages are up to date. Also contains fixes like spawning RStudio from the launcher / SageMath fixes compared to our original image.

### CODE REVIEW

File: repo2docker/Project.toml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[deps]
```
Comment (moorepants) [inline repo2docker/Project.toml:1]: These aren't private keys or anything are they?

File: repo2docker/apt.txt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+texlive-pstricks
+texlive-pstricks-doc
+texlive-publishers
+texlive-publishers-doc
+texlive-science
+texlive-science-doc
+texlive-xetex
+tipa
+vprerex
+xindy
```
Comment (moorepants) [inline repo2docker/apt.txt:80]: Here are some basic tools that I think we should include:

```
build-essential
clang
cmake
curl
ffmpeg
gfortran
git
htop
mercurial
subversion
nodejs
pandoc
tree
```

Or are there a set of default utils installed already?

**Comment by moorepants** [comment]
Looks good. Only comment would be to make sure we have apt packages that provide common core command line utilities.

File: repo2docker/Project.toml (hunk 1 above)
Comment (sandertyu) [reply repo2docker/Project.toml:1]: Within Julia, those keys are [basically the version number](https://julialang.github.io/Pkg.jl/v1/toml-files/#The-uuid-field-1). Nothing private to it.

File: repo2docker/apt.txt (hunk 2 above)
Comment (sandertyu) [reply repo2docker/apt.txt:80]: Added clang, cmake, ffmpeg, htop, pandoc and tree. Everything else listed there is already available.

**Comment by sandertyu** [comment]
Okay, all the code for this repo should now be set. We need to build the new image and make some changes to our helm values for this to be live. Once those are built and the FAQ is ready, we'll merge them all at once. The extension which adds an icon to the launcher and link to the FAQ in the "help" tab on JupyterLab uses our PostBuild, so they all need to be deployed at the same time to work properly.

**Comment by moorepants** [comment]
Sounds good. Can you add one more to the apt packages: `imagemagick`. Thanks.

**Comment by moorepants** [comment]
Nice work on this!

**Pull request merged**